### ISSUE TITLE
"Is Disjoint" rule is applied to CMTimeRange.intersection incorrectly

### ISSUE THREAD

**Issue by shosokawa** [realm/SwiftLint#3227]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
I got a warning for "Is Disjoint" #1796, even `Set` were not used. 
I think that this rule shouldn't be applied to `CMTimeRange`.

```swift
        if let audioSrcTrack = urlAsset.tracks(withMediaType: .audio).first,
            // swiftlint:disable is_disjoint
            !audioSrcTrack.timeRange.intersection(dstRange).isEmpty {
            // swiftlint:enable is_disjoint

            ...................................................

        }

```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
shMBP:Alpaca shosokawa$ Pods/SwiftLint/swiftlint lint
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'Constants.swift' (1/200)
Linting 'VideoExporter.swift' (2/200)
Linting 'UserNotificationHandler.swift' (3/200)
Linting 'GeocodeConverter.swift' (4/200)
Linting 'VideoComposer.swift' (5/200)
Linting 'Once.swift' (6/200)
Linting 'ReverseGeocoder.swift' (7/200)
Linting 'Logger.swift' (8/200)
Linting 'LRUCache.swift' (9/200)
Linting 'DebuggableOptionSet.swift' (10/200)
Linting 'VideoWriter.swift' (11/200)
Linting 'StatusBarMessages.swift' (12/200)
Linting 'Loader.swift' (13/200)
Linting 'ALPSynchronized.swift' (14/200)
Linting 'DirectionalPanGestureRecognizer.swift' (15/200)
Linting 'Array+Extensions.swift' (16/200)
Linting 'UIViewController+Extensions.swift' (17/200)
Linting 'DispatchQueue+Extensions.swift' (18/200)
Linting 'GestureRecognizer+AlwaysRecognizeDelegate.swift' (19/200)
Linting 'UIView+Extensions.swift' (20/200)
Linting 'DateFormatter+Extensions.swift' (21/200)
Linting 'String+Extensions.swift' (22/200)
Linting 'AVAsset+Extensions.swift' (23/200)
Linting 'UIFont+Extensions.swift' (24/200)
Linting 'MediaView+Extensions.swift' (25/200)
Linting 'UIColor+Extensions.swift' (26/200)
Linting 'UIImage+Extensions.swift' (27/200)
Linting 'AVFoundation+Extenions.swift' (28/200)
Linting 'UIApplication+Extensions.swift' (29/200)
Linting 'UIAlertController+Extensions.swift' (30/200)
Linting 'UIScreen+Extensions.swift' (31/200)
Linting 'NotificationCenter+registration.swift' (32/200)
Linting 'FileManager+Extensions.swift' (33/200)
Linting 'Calendar+Extensions.swift' (34/200)
Linting 'CALayer+Extensions.swift' (35/200)
Linting 'URL+Extensions.swift' (36/200)
Linting 'UUID+Extensions.swift' (37/200)
Linting 'UIScrollView+Extensions.swift' (38/200)
Linting 'UISize+square.swift' (39/200)
Linting 'Date+Extensions.swift' (40/200)
Linting 'Photos+Extensions.swift' (41/200)
Linting 'CGRect+Extensions.swift' (42/200)
Linting 'NavigationController+extensions.swift' (43/200)
Linting 'Collection+Extensions.swift' (44/200)
Linting 'UICollectionView+Extensions.swift' (45/200)
Linting 'UIView+rotate.swift' (46/200)
Linting 'CATransition+dismissModal.swift' (47/200)
Linting 'Tracker.swift' (48/200)
Linting 'SynchronizedArray.swift' (49/200)
Linting 'Notifications.swift' (50/200)
Linting 'BackgroundTask.swift' (51/200)
Linting 'AssetOperation.swift' (52/200)
Linting 'Cache.swift' (53/200)
Linting 'SynchronizedDictionary.swift' (54/200)
Linting 'Geohash.swift' (55/200)
Linting 'LocationManager.swift' (56/200)
Linting 'Color.swift' (57/200)
Linting 'PermissionAccess+Extensions.swift' (58/200)
Linting 'VideoImageList.swift' (59/200)
Linting 'UserLocation.swift' (60/200)
Linting 'ElementPlaybackItem.swift' (61/200)
Linting 'FirebaseError.swift' (62/200)
Linting 'Relation.swift' (63/200)
Linting 'Connection.swift' (64/200)
Linting 'Friend.swift' (65/200)
Linting 'FirestoreDocumentable.swift' (66/200)
Linting 'IndexedItem.swift' (67/200)
Linting 'FirebaseFunctions+Extensions.swift' (68/200)
Linting 'ALPNotification.swift' (69/200)
Linting 'Element.swift' (70/200)
Linting 'FirebaseAuth+Extensions.swift' (71/200)
Linting 'User.swift' (72/200)
Linting 'TextValidator.swift' (73/200)
Linting 'Reaction.swift' (74/200)
Linting 'VideoMetadata.swift' (75/200)
Linting 'TextMemory.swift' (76/200)
Linting 'Clip.swift' (77/200)
Linting 'Firestore+Extensions.swift' (78/200)
Linting 'Defaults.swift' (79/200)
Linting 'CalendarItem.swift' (80/200)
Linting 'Observable.swift' (81/200)
Linting 'FirebaseStorage+Extensions.swift' (82/200)
Linting 'TextFilterable.swift' (83/200)
Linting 'PlaybackSpan.swift' (84/200)
Linting 'Circle.swift' (85/200)
Linting 'Diffable.swift' (86/200)
Linting 'Belonging.swift' (87/200)
Linting 'FriendCircle.swift' (88/200)
Linting 'PhotoLibrary.swift' (89/200)
Linting 'VideoResourceGenerator.swift' (90/200)
Linting 'VideoImageListGenerator.swift' (91/200)
Linting 'FriendListener.swift' (92/200)
Linting 'ElementsPresenter.swift' (93/200)
Linting 'FriendCircleRequestListener.swift' (94/200)
Linting 'DocumentListener.swift' (95/200)
Linting 'PaginationListener.swift' (96/200)
Linting 'StorageObjectDownloader.swift' (97/200)
Linting 'FriendRequestListener.swift' (98/200)
Linting 'NearbyUsersPresenter.swift' (99/200)
Linting 'VideoImageListGenerateOperation.swift' (100/200)
Linting 'VideoAssetGenerator.swift' (101/200)
Linting 'VideoAssetGenerateOperation.swift' (102/200)
Linting 'UserLocationTracker.swift' (103/200)
Linting 'StorageObjectUploader.swift' (104/200)
Linting 'MemoriesPresenter.swift' (105/200)
Linting 'FriendCircleListener.swift' (106/200)
Linting 'FriendsPresenter.swift' (107/200)
Linting 'VideoAssetConvertOperation.swift' (108/200)
Linting 'SharedElementListener.swift' (109/200)
Linting 'NotificationListener.swift' (110/200)
Linting 'AppDelegate.swift' (111/200)
Linting 'CircleAvatarsView.swift' (112/200)
Linting 'SettingsTableViewController.swift' (113/200)
Linting 'InformationViewController.swift' (114/200)
Linting 'Poppable.swift' (115/200)
Linting 'FullScreenPlaybackCells.swift' (116/200)
Linting 'ActionBar.swift' (117/200)
Linting 'FriendsCollectionViewCell.swift' (118/200)
Linting 'JustTapGestureRecognizer.swift' (119/200)
Linting 'BlurredImageView.swift' (120/200)
Linting 'TextFields.swift' (121/200)
Linting 'Buttons.swift' (122/200)
Linting 'FriendsCollectionView.swift' (123/200)
Linting 'BackgroundViews.swift' (124/200)
Linting 'CalendarCells.swift' (125/200)
Linting 'Spinner.swift' (126/200)
Linting 'NavigationBar.swift' (127/200)
Linting 'CollectionViews.swift' (128/200)
Linting 'PickerButton.swift' (129/200)
Linting 'DestinationsCollectionView.swift' (130/200)
Linting 'CameraViewController.swift' (131/200)
Linting 'CameraButton.swift' (132/200)
Linting 'DestinationCell.swift' (133/200)
Linting 'Camera.swift' (134/200)
Linting 'FilteredPreviewView.swift' (135/200)
Linting 'BoomerangImageView.swift' (136/200)
Linting 'VideoOverlayView.swift' (137/200)
Linting 'PageControl.swift' (138/200)
Linting 'FullScreenPlaybackView.swift' (139/200)
Linting 'CenteredTextView.swift' (140/200)
Linting 'ViewControllers.swift' (141/200)
Linting 'ImageViews.swift' (142/200)
Linting 'GridCollectionViews.swift' (143/200)
Linting 'ProgressBar.swift' (144/200)
Linting 'Labels.swift' (145/200)
Linting 'MovieSlideView.swift' (146/200)
Linting 'CombinedLabelView.swift' (147/200)
Linting 'SignInViewController.swift' (148/200)
Linting 'PlaybackSpanPickerViewController.swift' (149/200)
Linting 'PlaybackSpanPickerView.swift' (150/200)
Linting 'PlaybackViewController.swift' (151/200)
Linting 'CalendarViewController.swift' (152/200)
Linting 'KeyboardConstraint.swift' (153/200)
Linting 'GridCells.swift' (154/200)
Linting 'PlayerView.swift' (155/200)
Linting 'FriendSearchBar.swift' (156/200)
Linting 'FlaggedPhoneNumberTextField.swift' (157/200)
Linting 'EmojiEmitter.swift' (158/200)
Linting 'FriendsViewController.swift' (159/200)
Linting 'FriendCircleNavigationBar.swift' (160/200)
Linting 'FriendCircleDayViewController.swift' (161/200)
Linting 'FriendCell.swift' (162/200)
Linting 'FriendCircleCreatorViewController.swift' (163/200)
Linting 'ElementPlaybackViewController.swift' (164/200)
Linting 'EmojiPickerView.swift' (165/200)
Linting 'FriendCircleCalendarViewController.swift' (166/200)
Linting 'FriendCircleRequestViewController.swift' (167/200)
Linting 'FriendCircleMemberCell.swift' (168/200)
Linting 'AvatarImageView.swift' (169/200)
Linting 'FriendRequestViewController.swift' (170/200)
Linting 'PhoneNumberSearchResultViewController.swift' (171/200)
Linting 'ColorCell.swift' (172/200)
Linting 'FriendSearchResultCell.swift' (173/200)
Linting 'ColorPickerViewController.swift' (174/200)
Linting 'AvatarSettingsController.swift' (175/200)
Linting 'FriendSearchView.swift' (176/200)
Linting 'AccountViewController.swift' (177/200)
Linting 'FriendSearchViewController.swift' (178/200)
Linting 'PhoneNumberSearchViewController.swift' (179/200)
Linting 'NameEditorViewController.swift' (180/200)
Linting 'LocationSettingsViewController.swift' (181/200)
Linting 'PhotoLibrarySettingsViewController.swift' (182/200)
Linting 'DayCells.swift' (183/200)
Linting 'NotificationBadgeButton.swift' (184/200)
Linting 'TextMemoryEditorViewController.swift' (185/200)
Linting 'AccountButton.swift' (186/200)
Linting 'MemoriesViewController.swift' (187/200)
Linting 'TodayButton.swift' (188/200)
Linting 'MemoryDetailViewController.swift' (189/200)
Linting 'DayMenuBar.swift' (190/200)
Linting 'ClipPlaybackViewController.swift' (191/200)
Linting 'MemoriesNavigationController.swift' (192/200)
Linting 'ElementPickerViewController.swift' (193/200)
Linting 'ClipPreviewViewController.swift' (194/200)
Linting 'ClipPickerViewController.swift' (195/200)
Linting 'FriendPickerViewController.swift' (196/200)
Linting 'FriendPickerCell.swift' (197/200)
Linting 'FriendPickerCollectionView.swift' (198/200)
Linting 'UIWindow+Debug.swift' (199/200)
Linting 'DebugPanelViewController.swift' (200/200)
/Users/shosokawa/Documents/episodes/git_repo/onesec/client/ios/Alpaca/Alpaca/Presenters/VideoAssetConvertOperation.swift:135:38: warning: Is Disjoint Violation: Prefer using `Set.isDisjoint(with:)` over `Set.intersection(_:).isEmpty`. (is_disjoint)
Done linting! Found 1 violation, 0 serious in 200 files.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.39.2
* Installation method used 
CocoaPods
* Paste your configuration file:

```yml
disabled_rules:
  - force_cast
  - force_try
  - identifier_name
  - file_length
  - function_body_length
  - shorthand_operator
  - todo
  - discarded_notification_center_observer
  - cyclomatic_complexity
  - nesting
  - large_tuple
  - unused_setter_value
  - multiple_closures_with_trailing_closure

opt_in_rules:
  # - empty_count
  - explicit_init
  - closure_spacing
  - closure_end_indentation
  - operator_usage_whitespace

line_length: 180

type_body_length:
  warning: 300
  error: 400

included:
  - Alpaca

excluded:
  - Pods
  - Alpaca/Supporting Files/R.generated.swift

custom_rules:
  indentation:
    name: "Indentation"
    message: "Tab is not allowed."
    regex: "\t"
    severity: error
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
No.
* Which Xcode version are you using (check `xcodebuild -version`)?
Xcode 11.5
Build version 11E608c
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```
echo 'let range = CMTimeRange(start: .zero, duration: .zero); _ = !range.intersection(range).isEmpty' | swiftlint lint --no-cache --use-stdin --enable-all-rules
```
